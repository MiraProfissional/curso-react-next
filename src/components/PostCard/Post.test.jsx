import { render, screen } from "@testing-library/react";
import { PostCard } from '.';

const mock = {
    title: 'title teste',
    body: 'body teste',
    id: 1,
    cover: 'img/img.png'
}

describe('<PostCard />', () => {

    it('should render PostCard correctly', () => {

        render(<PostCard {...mock} />);
        
        expect(screen.getByRole('img', { name: /title teste/i })).toHaveAttribute('src', 'img/img.png');
        expect(screen.getByRole('heading', { name: 'title teste 1' })).toBeInTheDocument();
        expect(screen.getByText('body teste')).toBeInTheDocument();
    })

    it('should match snapshot', () => {

        const { container } = render(<PostCard {...mock} />);
        expect(container.firstChild).toMatchSnapshot();
        
    })

})