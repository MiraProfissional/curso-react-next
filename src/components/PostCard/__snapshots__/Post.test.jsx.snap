// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div
  class="post-card"
>
  <img
    alt="title teste"
    src="img/img.png"
  />
  <div
    class="post-content"
  >
    <h2>
      title teste
       
      1
    </h2>
    <p>
      body teste
    </p>
  </div>
</div>
`;
